packages/cli/config/jest.js

```diff
@@ -131,10 +131,38 @@ async function getProjectConfig(targetPath, displayName) {
     },
 
     transform: {
-      '\\.(js|jsx|ts|tsx|mjs|cjs)$': [
+      '\\.(mjs|cjs)$': [
         require.resolve('./jestSwcTransform'),
         {
           sourceMaps: envOptions.enableSourceMaps || envOptions.nextTests,
+          jsc: {
+            target: 'es2019',
+            parser: {
+              syntax: 'ecmascript',
+            },
+            transform: {
+              react: {
+                runtime: 'automatic',
+              },
+            },
+          },
+        },
+      ],
+      '\\.(t|j)sx?$': [
```

Comment:
Probably better to split this into two blocks I think? don't think it's safe to use the typescript parser for JSX